docs/users_guide/safe_haskell.rst

```diff
@@ -273,11 +273,6 @@ Furthermore, we restrict the following features:
   this reason, the ``Data.Coerce`` module is also considered unsafe. We are
   hoping to find a better solution here in the future.
 
-- ``Data.Typeable`` — Hand crafted instances of the Typeable type class are not allowed
-  in Safe Haskell as this can easily be abused to unsafely coerce
-  between types. Derived instances (through the :ghc-flag:`-XDeriveDataTypeable`
-  extension) are still allowed.
-
```

Comment:
Correct. The only way to (sort of) get a hand-written `Typeable` is using extremely unsafe deep magic, `reflection` style. And even that won't give you a global instance.